akka-actor-typed/src/main/scala/akka/actor/typed/TypedActorContext.scala

```diff
@@ -25,4 +27,19 @@ trait TypedActorContext[T] {
    * Get the `scaladsl` of this `ActorContext`.
    */
   def asScala: scaladsl.ActorContext[T]
+
+  /**
+   * Narrow the type of this `TypedActorContext`, which is always a safe operation.
+   */
+  def narrow[U <: T]: TypedActorContext[U]
+
+  /**
+   * Unsafe utility method for widening the type accepted by this TypedActorContext.
+   * */
+  def unsafeUpcast[U >: T @uncheckedVariance]: TypedActorContext[U]
+
+  /**
+   * Unsafe utility method for changing the type accepted by this TypedActorContext.
+   * */
+  @InternalApi private[akka] def unsafeCast[U]: TypedActorContext[U]
```

Comment:
that hides a bit of ugliness in the internals of Akka

The doc comment is wrong:
```
   /**
    * INTERNAL API: Unsafe utility method for changing the type accepted by this TypedActorContext.
    */
```